pkg/generate/app/cmd/newapp.go

```diff
@@ -587,14 +591,23 @@ func ensureValidUniqueName(names map[string]int, name string) (string, error) {
 	if len(name) < 2 {
 		return "", fmt.Errorf("invalid name: %s", name)
 	}
+
+	// Make all names lowercase
+	name = strings.ToLower(name)
+
+	// Remove everything except [-0-9a-z]
+	name = invalidNameCharactersRegexp.ReplaceAllString(name, "")
+
+	// Remove leading hyphen (that may be introduced by the previous step)
+	if strings.HasPrefix(name, "-") {
```

Comment:
this if is unnecessary, just TrimLeft, if there is no "-" character on left side, it will be no-op.